pytrickle/api_spec.py

```diff
@@ -0,0 +1,70 @@
+"""
+Core Pydantic models for Trickle streaming API parameters.
+
+This module defines reusable BaseModel classes for validating and serializing
+API request parameters used across trickle streaming applications.
+"""
+
+import json
+from typing import Any, Dict, List, Optional, Union
+from pydantic import BaseModel, Field, field_validator
+
+class StreamStartRequest(BaseModel):
+    """Base request model for starting a trickle stream."""
+    subscribe_url: str = Field(..., description="URL for subscribing to input video stream")
+    publish_url: str = Field(..., description="URL for publishing output video stream")
+    control_url: Optional[str] = Field(default=None, description="URL for control channel communication")
+    events_url: Optional[str] = Field(default=None, description="URL for events channel communication")
+    data_url: Optional[str] = Field(default=None, description="URL for publishing text/data output via data channel")
+    gateway_request_id: str = Field(..., description="Unique identifier for the stream request")
+    
+    # Optional fields that may be present in the request
+    manifest_id: Optional[str] = Field(default=None, description="Manifest identifier")
+    model_id: Optional[str] = Field(default=None, description="Model identifier")
+    stream_id: Optional[str] = Field(default=None, description="Stream identifier")
+
+class StreamParamsUpdateRequest(BaseModel):
```

Comment:
This should be more dynamic I think.

Maybe just `params` object that is a `Dict[Any]`?

Specifically for width and height, it its hard to change that mid stream right?